### PR TITLE
Provide better gjs/gts config support for eslint 9

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
Resolves: https://github.com/ember-cli/eslint-plugin-ember/issues/2190
Resolves: https://github.com/ember-cli/eslint-plugin-ember/issues/2171 (sorta)

Unblocks: https://github.com/ember-cli/ember-cli/pull/10533

### CODE REVIEW

File: lib/recommended.mjs
Comment (NullVoxPopuli) [inline lib/recommended.mjs:1]: new public import, only available via import, not require

**Review by patricklx — APPROVED** [review]
any chance to add a test?

**Review by wagenet — APPROVED** [review]

**Comment by NullVoxPopuli** [comment]
> any chance to add a test?

I think we'd have to add a whole acceptance suite -- which I'm a fan of -- and we used to have, but I think @bmish removed (because acceptance tests are slow).


_I_ think it's valuable enough where we should _just do it_, and pay the cost of acceptance tests.

Is everyone ok with the tradeoff? 
It would be great to not have to have the slow feedback loop with ember-cli's blueprint system to test changes to the configs in this project.

**Comment by NullVoxPopuli** [comment]
Gonna merge -- acceptance testing infra is too big to lump in to this PR and has some decisions that may need discussing / education / alignment / etc / whatever

**Pull request merged**